packages/clay-form/src/MultiSelect.tsx

```diff
@@ -0,0 +1,319 @@
+	deliminatorKeys?: number[];
+
+	/**
+	 * Message to display if validation fails.
+	 */
+	errorMessage?: string;
+
+	/**
+	 * Ref added to container element
+	 */
+	forwardRef: React.RefObject<HTMLDivElement>;
+
+	/**
+	 * Message to display which helps users interact with input
+	 */
+	helpText?: string;
+
+	/**
+	 * Value of input
+	 */
+	inputValue?: string;
+
+	/**
+	 * Values that display as label items
+	 */
+	items: string[];
+
+	/**
+	 * Description of the input
+	 */
+	label?: React.ReactText;
+
+	/**
+	 * Callback for when items are added or removed
+	 */
+	onItemsChange: (val: string[]) => void;
+
+	/**
+	 * Callback for when the input value changes
+	 */
+	onInputChange?: (val: string) => void;
+
+	/**
+	 * List of pre-populated items that will show up in a dropdown menu
+	 */
+	sourceItems?: string[];
+
+	/**
+	 * Path to spritemap for clay icons
+	 */
+	spritemap?: string;
+
+	/**
+	 * Flag to indicate if validation should occur when input is blured
+	 */
+	validateOnBlur?: boolean;
+
+	/**
+	 * Callback for validating item. Returning `true` means the value is valid.
+	 */
+	validationFn?: (val?: string) => boolean;
+}
+
+const MultiSelect: React.FunctionComponent<IProps> = ({
+	deliminatorKeys = KEYS,
+	errorMessage,
+	forwardRef,
+	helpText,
+	items,
+	label,
+	onItemsChange,
+	spritemap,
+	validateOnBlur = false,
+	validationFn = () => true,
+	inputValue = '',
+	onInputChange,
+	sourceItems,
+	...otherProps
+}) => {
+	const inputRef = useRef<HTMLInputElement>(null);
+	const lastItemRef = useRef<HTMLSpanElement>(null);
+	const [active, setActive] = useState(false);
+	const [isFocused, setIsFocused] = useState();
+	const [isValid, setIsValid] = useState(true);
+
+	useLayoutEffect(() => {
+		if (sourceItems) {
+			const matchedItems = sourceItems.filter(
+				item => inputValue && item.match(inputValue)
+			);
+
+			setActive(matchedItems.length !== 0);
+		}
+	}, [sourceItems, inputValue]);
+
+	const setNewValue = (newVal: string) => {
+		if (validationFn(newVal)) {
+			onItemsChange([...items, newVal]);
+
+			if (onInputChange) {
+				onInputChange('');
+			}
+		} else {
+			setIsValid(false);
+		}
+	};
+
+	const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+		const {keyCode} = event;
+
+		if (inputValue && deliminatorKeys.includes(keyCode)) {
+			event.preventDefault();
+
+			setNewValue(inputValue);
+		} else if (
+			!inputValue &&
+			keyCode === BACKSPACE_KEY &&
+			inputRef.current &&
+			lastItemRef.current
+		) {
+			inputRef.current.blur();
+			lastItemRef.current.focus();
+		} else {
+			setIsValid(true);
+		}
+	};
+
+	const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+		const {value} = event.target;
+
+		setIsFocused(false);
+
+		if (value && validateOnBlur) {
+			setNewValue(value);
+		}
+	};
+
+	const handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+		const pastedText = event.clipboardData.getData('Text');
+
+		const pastedItems = pastedText
+			.split(',')
```

Comment:
Seems this should be the same as the "deliminators", right?

Reply:
Hmm, good catch. Not sure it would be the same since characters and keyCodes aren't necessarily the same thing. Let me think about this